src/main/java/org/wso2/config/mapper/ReferenceResolver.java

```diff
@@ -111,8 +113,10 @@ private static void resolveConfigReferences(Map<String, Object> context) throws
                         extractPlaceholdersFromString(key, (String) elementItem, unresolvedKeys, valuesToResolve);
                     }
                 }
+            } else if (value instanceof Map) {
+                resolveConfigReferences((Map<String, Object>) value);
```

Comment:
Will this work as we are not passing the full context?

Reply:
Reverted as this will needs to be addressed in a separate issue.